R/snapshot-reporter.R

```diff
@@ -101,6 +103,19 @@ SnapshotReporter <- R6::R6Class("SnapshotReporter",
       } else {
         snap_dir <- file.path(self$snap_dir, variant, self$file)
       }
+
+      save_path <- file.path(snap_dir, name)
+      if (save_path %in% self$snap_file_saved) {
+        fail(
```

Comment:
Should this be an error rather than a failure? I think it represents a bad call to `expect_snapshot_file()` rather than a failing test? Am I thinking about this correctly?

Reply:
Switching to `cli::cli_abort("... {.arg name} ...)`